src/routes/notifications/notifications.controller.spec.ts

```diff
@@ -0,0 +1,243 @@
+    fakeConfigurationService.set(
+      'exchange.apiKey',
+      faker.random.alphaNumeric(),
+    );
+  });
+
+  beforeEach(async () => {
+    jest.clearAllMocks();
+    const moduleFixture: TestingModule = await Test.createTestingModule({
+      imports: [
+        // feature
+        NotificationsModule,
+        // common
+        DomainModule,
+        TestCacheModule,
+        TestConfigurationModule,
+        TestNetworkModule,
+      ],
+    }).compile();
+
+    app = moduleFixture.createNestApplication();
+    app.useGlobalFilters(new DataSourceErrorFilter());
+
+    await app.init();
+  });
+
+  describe('POST /register/notifications', () => {
+    it('Success', async () => {
+      const chains: Chain[] = range(4).map((idx) =>
+        chainBuilder().with('chainId', idx.toString()).build(),
+      );
+      const safeRegistrations: SafeRegistration[] = chains.map((chain) =>
+        safeRegistrationBuilder().with('chain_id', chain.chainId).build(),
+      );
+      const registerDeviceDto = registerDeviceDtoBuilder()
+        .with('safeRegistration', safeRegistrations)
+        .build();
+      mockNetworkService.get.mockImplementation((url) => {
+        const getChainUrlPattern = `${safeConfigUrl}/api/v1/chains/`;
+        if (url.includes(getChainUrlPattern)) {
+          return Promise.resolve({ data: sample(chains) });
+        }
+        return Promise.reject(`No matching rule for url: ${url}`);
+      });
+      mockNetworkService.post.mockImplementation(() => Promise.resolve());
+
+      await request(app.getHttpServer())
+        .post('/register/notifications')
+        .send(registerDeviceDto)
+        .expect(200);
+    });
+
+    it('Client errors returned from provider', async () => {
+      const chains: Chain[] = range(4).map((idx) =>
+        chainBuilder().with('chainId', idx.toString()).build(),
+      );
+      const safeRegistrations: SafeRegistration[] = chains.map((chain) =>
+        safeRegistrationBuilder().with('chain_id', chain.chainId).build(),
+      );
+      const registerDeviceDto = registerDeviceDtoBuilder()
+        .with('safeRegistration', safeRegistrations)
+        .build();
+      mockNetworkService.get.mockImplementation((url) => {
+        const getChainUrlPattern = `${safeConfigUrl}/api/v1/chains/`;
+        if (url.includes(getChainUrlPattern)) {
+          return Promise.resolve({ data: sample(chains) });
+        }
+        return Promise.reject(`No matching rule for url: ${url}`);
+      });
+      // return a client error for just one provider
+      const error = new NetworkResponseError(
+        faker.datatype.number({ min: 400, max: 499 }),
+      );
+      mockNetworkService.post
+        .mockImplementationOnce(() => Promise.reject(error))
+        .mockImplementation(() => Promise.resolve());
+
+      await request(app.getHttpServer())
+        .post('/register/notifications')
+        .send(registerDeviceDto)
+        .expect(400);
```

Comment:
What's the expectation of the body of the message in this case?

(same for the other tests)

Reply:
True, the body in case of error is missing, I will change this part, thanks!

Reply:
I changed the approach in https://github.com/5afe/safe-client-gateway-nest/pull/267/commits/0728521aa558fa73c90498d9c13798e313936c93 

Now the expected body is checked. From [the current Rust implementation](https://github.com/safe-global/safe-client-gateway/blob/ad6e9f132e804fbacbbecbc878cb8b7efe4d2dc1/src/routes/notifications/handlers.rs#L90), I took the `message` only. Do you think the `code`, `arguments`, and `debug` parts are necessary as well? I remember this being discussed to not expose this `debug` data in the service responses, what do you think @fmrsabino?

Reply:
Let's keep it as you have right now. A change at this level would also impact other error payloads.